src/votekit/ballot_generator.py

```diff
@@ -1932,53 +1932,102 @@ def generate_profile(
+        voter_dist (Callable[..., np.ndarray]): Distribution to sample a single
+        voter's position from, defaults to uniform distribution.
+        voter_params: (dict[str, float], optional): Parameters to be passed to
+        voter_dist, defaults to uniform(0,1) in 2 dimensions.
+        candidate_dist: (Callable[..., np.ndarray]): Distribution to sample a
+        single candidate's position from, defaults to uniform distribution.
+        candidate_params: Optional[Dict[str, float]]: Parameters to be passed
+        to candidate_dist, defaults to uniform(0,1) in 2 dimensions.
+        distance: (Callable[[np.ndarray, np.ndarray], float]], optional):
+        Computes distance between a voter and a candidate,
+        defaults to euclidean distance.
 
     """
 
-    def generate_profile(
+    def __init__(
         self,
-        number_of_ballots: int,
-        by_bloc: bool = False,
-        dim: int = 2,
-        lower: float = 0,
-        upper: float = 1,
-        seed: Optional[int] = None,
+        candidates: list[str],
+        voter_dist: Callable[..., np.ndarray] = np.random.uniform,
+        voter_params: Optional[Dict[str, float]] = None,
+        candidate_dist: Callable[..., np.ndarray] = np.random.uniform,
+        candidate_params: Optional[Dict[str, float]] = None,
+        distance: Callable[[np.ndarray, np.ndarray], float] = euclidean_distance,
+    ):
+        super().__init__(candidates=candidates)
+        self.voter_dist = voter_dist
+        self.candidate_dist = candidate_dist
+
+        if voter_params is None:
+            # default params used for np.random.uniform
+            self.voter_params = {"low": 0.0, "high": 1.0, "size": 2.0}
+        else:
+            self.voter_params = voter_params
+
+        if candidate_params is None:
+            # default params used for np.random.uniform
+            self.candidate_params = {"low": 0.0, "high": 1.0, "size": 2.0}
+        else:
+            self.candidate_params = candidate_params
+
+        self.distance = distance
+
+    def generate_profile(
+        self, number_of_ballots: int, by_bloc: bool = False, seed: Optional[int] = None
     ) -> Union[PreferenceProfile, Tuple]:
         """
         Args:
             number_of_ballots (int): The number of ballots to generate.
             by_bloc (bool): Dummy variable from parent class.
-            dim (int, optional): number of dimensions to use, defaults to 2d
-            lower (float, optional): lower bound for uniform distribution, defaults to 0
-            upper (float, optional): upper bound for uniform distribution, defaults to 1
-            seed (int, optional): seed for random generation
+            seed (int, optional): Seed for random generation, defaults to None
+            which resets the random seed.
 
         Returns:
-            Union[PreferenceProfile, Tuple]
+            preference profile, candidate positions, voter positions:
```

Comment:
Correct return string syntax is `Union[PreferenceProfile, Tuple]: explanation of what is returned`

Reply:
@peterrrock2 , I'm just making a note that I think we can clean up how the `BallotGenerator` class is defined to avoid these weird type issues. The reason this is a Union[PrefProfile, Tuple] is b/c that's what the abstract method expects.